scripts/build/autotools/CheckSSL.m4

```diff
@@ -0,0 +1,240 @@
+      PHP_ADD_LIBRARY([tls],,[MONGODB_SHARED_LIBADD])
+      PHP_ADD_LIBRARY([crypto],,[MONGODB_SHARED_LIBADD])
+      PHP_MONGODB_SSL="libressl"
+      found_libressl="yes"
+    fi
+  fi
+
+  if test "$PHP_MONGODB_SSL" = "libressl" -a "$found_libressl" != "yes"; then
+    AC_MSG_ERROR([LibreSSL libraries and development headers could not be found])
+  fi
+])
+
+AS_IF([test "$PHP_MONGODB_SSL" = "darwin" -o \( "$PHP_MONGODB_SSL" = "auto" -a "$os_darwin" = "yes" \)],[
+  dnl PHP_FRAMEWORKS is only used for SAPI builds, so use MONGODB_SHARED_LIBADD for shared builds
+  if test "$ext_shared" = "yes"; then
+    MONGODB_SHARED_LIBADD="-framework Security -framework CoreFoundation $MONGODB_SHARED_LIBADD"
+  else
+    PHP_ADD_FRAMEWORK([Security])
+    PHP_ADD_FRAMEWORK([CoreFoundation])
+  fi
+  PHP_MONGODB_SSL="darwin"
+])
+
+AS_IF([test "$PHP_MONGODB_SSL" = "auto"],[
+  PHP_MONGODB_SSL="no"
+])
+
+AC_MSG_CHECKING([which TLS library to use])
+AC_MSG_RESULT([$PHP_MONGODB_SSL])
+
+dnl Disable Windows SSL and crypto
+AC_SUBST(MONGOC_ENABLE_SSL_SECURE_CHANNEL, 0)
+AC_SUBST(MONGOC_ENABLE_CRYPTO_CNG, 0)
+
+if test "$PHP_MONGODB_SSL" = "openssl" -o "$PHP_MONGODB_SSL" = "libressl" -o "$PHP_MONGODB_SSL" = "darwin"; then
+   AC_SUBST(MONGOC_ENABLE_SSL, 1)
+   AC_SUBST(MONGOC_ENABLE_CRYPTO, 1)
+   if test "$PHP_MONGODB_SSL" = "darwin"; then
+      AC_SUBST(MONGOC_ENABLE_SSL_OPENSSL, 0)
+      AC_SUBST(MONGOC_ENABLE_SSL_LIBRESSL, 0)
+      AC_SUBST(MONGOC_ENABLE_SSL_SECURE_TRANSPORT, 1)
+      AC_SUBST(MONGOC_ENABLE_CRYPTO_LIBCRYPTO, 0)
+      AC_SUBST(MONGOC_ENABLE_CRYPTO_COMMON_CRYPTO, 1)
+   elif test "$PHP_MONGODB_SSL" = "openssl"; then
+      AC_SUBST(MONGOC_ENABLE_SSL_OPENSSL, 1)
+      AC_SUBST(MONGOC_ENABLE_SSL_LIBRESSL, 0)
+      AC_SUBST(MONGOC_ENABLE_SSL_SECURE_TRANSPORT, 0)
+      AC_SUBST(MONGOC_ENABLE_CRYPTO_LIBCRYPTO, 1)
+      AC_SUBST(MONGOC_ENABLE_CRYPTO_COMMON_CRYPTO, 0)
+   elif test "$PHP_MONGODB_SSL" = "libressl"; then
+      AC_SUBST(MONGOC_ENABLE_SSL_OPENSSL, 0)
+      AC_SUBST(MONGOC_ENABLE_SSL_LIBRESSL, 1)
+      AC_SUBST(MONGOC_ENABLE_SSL_SECURE_TRANSPORT, 0)
+      AC_SUBST(MONGOC_ENABLE_CRYPTO_LIBCRYPTO, 1)
+      AC_SUBST(MONGOC_ENABLE_CRYPTO_COMMON_CRYPTO, 0)
+   fi
+else
+   AC_SUBST(MONGOC_ENABLE_SSL, 0)
+   AC_SUBST(MONGOC_ENABLE_SSL_LIBRESSL, 0)
+   AC_SUBST(MONGOC_ENABLE_SSL_OPENSSL, 0)
+   AC_SUBST(MONGOC_ENABLE_SSL_SECURE_TRANSPORT, 0)
+   AC_SUBST(MONGOC_ENABLE_CRYPTO, 0)
+   AC_SUBST(MONGOC_ENABLE_CRYPTO_LIBCRYPTO, 0)
+   AC_SUBST(MONGOC_ENABLE_CRYPTO_COMMON_CRYPTO, 0)
+fi
+
+if test "$PHP_MONGODB_SSL" = "openssl"; then
+  old_CFLAGS="$CFLAGS"
+  CFLAGS="$PHP_MONGODB_SSL_CFLAGS"
+
+  AC_CHECK_DECL([ASN1_STRING_get0_data],
+                [AC_SUBST(MONGOC_HAVE_ASN1_STRING_GET0_DATA, 1)],
+                [AC_SUBST(MONGOC_HAVE_ASN1_STRING_GET0_DATA, 0)],
+                [[#include <openssl/asn1.h>]])
+
+  CFLAGS="$old_CFLAGS"
+else
+  AC_SUBST(MONGOC_HAVE_ASN1_STRING_GET0_DATA, 0)
```

Comment:
You've already done the ASN1_STRING_get0_data check earlier. Do you need it twice?

Reply:
I believe I forgot to remove this if/else block from the file. The problem with this is that `CFLAGS` may change depending on whether we found OpenSSL with `pkg-config` (sets `PHP_MONGODB_SSL_CFLAGS`) or manual header searching and `PHP_CHECK_LIBRARY` (sets `OPENSSL_INCDIR`).

Therefore, this if/else block should be removed and we should only keep the one below here:

```m4
if test "x$have_ASN1_STRING_get0_data" = "xyes"; then
   AC_SUBST(MONGOC_HAVE_ASN1_STRING_GET0_DATA, 1)
else
   AC_SUBST(MONGOC_HAVE_ASN1_STRING_GET0_DATA, 0)
fi
```